### PR TITLE
Devise Token Auth for one quran

### PR DESCRIPTION
**Pull request by mmahalwy**

### CODE REVIEW

**Comment by ahmedre** [comment]
Deployed to: http://staging.quran.com:33119

File: src/components/FacebookTokenButton/index.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+const FacebookTokenButton = ({ save, push }) => { // eslint-disable-line
+  let popup = null;
+  let interval = null;
+
+  const handleClick = () => {
+    popup = window.open('/onequran/omniauth/facebook?omniauth_window_type=newWindow&resource_class=User', '_blank'); // eslint-disable-line
+    interval = setInterval(() => popup.postMessage('requestCredentials', '*'), 1000);
+
+    window.addEventListener('message', (event) => { // eslint-disable-line
```
Comment (sabeurthabti) [inline src/components/FacebookTokenButton/index.js]: maybe a less generic event name?

File: src/redux/modules/auth.js [hunk 2]
```diff
@@ -14,6 +15,22 @@ const initialState = {
 
 export default function reducer(state = initialState, action = {}) {
   switch (action.type) {
+    case SAVE: {
+      cookie.save('auth', {
+        client: action.data.client_id,
+        'access-token': action.data.auth_token,
+        expiry: action.data.expiry,
+        uid: action.data.uid,
+        'token-type': 'Bearer'
```
Comment (sabeurthabti) [inline src/redux/modules/auth.js]: alignment of these properties.

File: src/utils/checkValidSurah.js [hunk 3]
```diff
@@ -1,9 +1,9 @@
 export default function isValidSurah(nextState, replaceState) {
   const surahId = parseInt(nextState.params.surahId, 10);
 
-  if (isNaN(surahId) || surahId > 114 || surahId < 1) {
-    replaceState('/error/invalid-surah');
-  }
+  // if (isNaN(surahId) || surahId > 114 || surahId < 1) {
```
Comment (sabeurthabti) [inline src/utils/checkValidSurah.js]: why is this removed?

File: src/components/FacebookTokenButton/index.js (hunk 1 above)
Comment (mmahalwy) [inline src/components/FacebookTokenButton/index.js]: That's the `postMessage` event name :(

File: src/utils/checkValidSurah.js (hunk 3 above)
Comment (mmahalwy) [inline src/utils/checkValidSurah.js]: ooops!

**Comment by ahmedre** [comment]
Deployed to: http://staging.quran.com:33122

**Comment by ahmedre** [comment]
Deployed to: http://staging.quran.com:33124
